hysr/main_sim.py

```diff
@@ -0,0 +1,207 @@
+        Generate a trajectory using the trajectory_getter (cf constructor)
+        and load this trajecty to the mujoco backend. Note that
+        as the mujoco backend is running in bursting mode, the trajectory
+        will not start playing until the burst method of the handle is
+        called.
+        """
+        trajectory = self._trajectory_getter.get_one()
+        rate = self._trajectory_getter.get_sample_rate()
+        rate_ns = int(rate * 1e9)
+        duration = o80.Duration_us.nanoseconds(rate_ns)
+
+        # going to first trajectory point
+        self._frontend_ball.add_command(
+            trajectory[0].position, trajectory[0].velocity, o80.Mode.OVERWRITE
+        )
+        # loading full trajectory
+        for item in trajectory[1:]:
+            self._frontend_ball.add_command(
+                item.position, item.velocity, duration, o80.Mode.QUEUE
+            )
+        self._frontend_ball.pulse()
+
+    def reset(self) -> None:
+        """
+        Do a full simulation reset, i.e. restore the state of the 
+        first simulation step, where all items are set according
+        to the mujoco xml configuration file.
+        """
+        self._handle.reset(_ball_segment_id)
+
+    def get_contact(self) -> context.ContactInformation:
+        """
+        Return the contact information between the ball and the racket.
+        The returned instance has the attributes:
+        - contact_occured : if true, at least one contact has occured
+        - position: if contact occured, the 3d position of the first contact
+        - time_stamp: if contact occured, the time stamp of the fist contact
+        - minimal_distance: if contact did not occure, the minimal distance
+                            between the two items
+        - disabled : true if contact detection has been disabled
+        Note that once a contact occured, the ball is no longer controlled by
+        o80 (i.e. the load method of this class will have no effect for the
+        corresponding ball), but by mujoco engine (until the method
+        reset_contacts of this class is called)
+        """
+
+        return self._handle.get_contact(_ball_segment_id)
+
+    def activate_contact(self, index: types.ListOrIndex = None) -> None:
+        """
+        Contact will no longer be ignored (if 'deactivate_contact'
+        has been previously called)
+        """
+        self._handle.activate_contact(_ball_segment_id)
+
+    def deactivate_contact(self) -> None:
+        """
+        Contacts between the ball and the racket
+        will be ignored.
+        """
+        self._handle.deactivate_contact(_ball_segment_id)
+
+    def reset_contact(self) -> None:
+        """
+        Reset the contact between the ball and the racket,
+        i.e. past contacts will be 'forgotten' and the 
+        o80 backend regains control of the ball.
+        """
+        self._handle.reset_contact(_ball_segment_id)
+
+    def get(self) -> types.MainSimState:
```

Comment:
Just `get` is a bit vague, maybe something like `get_state` would be better?

Reply:
updated !